### DISCUSSION TITLE
OTA flashing GAP8 with other decks connected

### DISCUSSION THREAD

**Discussion by machacekch** [bitcraze/discussions#1436] [Q&A]
I can't flash the AI deck's GAP8 over the air when there's other decks connected.
As noted in #1400 a LH deck gives an error, but even just the Qi deck makes flashing stop at 6%.
For reference, the command I'm running is `cfloader flash some.img deck-bcAI:gap8-fw -w radio://0/80/2M` and when removing the other decks it works just fine.

Is this a software or hardware issue? Having to take the cf2 apart to flash is extremely inconvenient, especially since the deck connectors are very fragile and won't survive repeated reassembly - the plastic's already starting to break.

**Comment by gemenerik** [comment]
I understand that removing decks for flashing is not a sustainable solution. 

Can you retry while running the latest master branch of the crazyflie-firmware? Commit https://github.com/bitcraze/crazyflie-firmware/commit/c2b949702e2a64017f769ad52f39c952359b5cf7 should fix this.

**Reply by machacekch** [reply]
> I understand that removing decks for flashing is not a sustainable solution.
> 
> Can you retry while running the latest master branch of the crazyflie-firmware? Commit [bitcraze/crazyflie-firmware@c2b9497](https://github.com/bitcraze/crazyflie-firmware/commit/c2b949702e2a64017f769ad52f39c952359b5cf7) should fix this.

Thanks for investigating. This did not fix the issue for me. Now flashing gets stuck at 3%:

```
aideck-gap8-examples$ cfloader flash examples/other/wifi-img-streamer/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img deck-bcAI:gap8-fw -w radio://0/80/2M
nRF51 has: sd-s130 and requires None and upgrade provides None. Current bootloader version is [2024.2.0] but upgrade provides [None]
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcLighthouse4, not in the target list
Skipping bcAI:esp, not in the target list
Deck bcAI:gap8, reset to bootloader
| 0% Writing to bcAI:gap8 deck memory
/ 1% Writing to bcAI:gap8 deck memory
- 2% Writing to bcAI:gap8 deck memory
\ 3% Writing to bcAI:gap8 deck memory
```

Could you please confirm that the only change I have to make in order to apply your fix is updating the stm32 firmware as follows:

```
crazyflie-firmware$ cfloader flash build/cf2.bin stm32-fw -w radio://0/80/2M
```

...since that's what I did. The startup log then looks like this:

```
---
SYS: Crazyflie 2.1 is up and running!
SYS: Build 158:079f02ed8f4d (2023.11 +158) CLEAN
SYS: I am 0x2030355451315006002C0058 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_INFO: Warning! No driver found for deck.
DECK_CORE: 3 deck(s) found
DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 0
DECK_CORE: Calling INIT on driver bcAI for deck 2
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized, CPX_VERSION: 0
AIDECK: Not setting up WiFi
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 2 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 10728 bytes
LHFL: Lighthouse bootloader version: 2
CPX: ESP32: I (434) SYS: Initialized
CPX: ESP32: I (434) DISCOVERY: Hostname is aideck-7BBDE8
CPX: ESP32: I (434) DISCOVERY: MAC (STA) is 78:21:84:7b:bd:e8
STAB: Starting stabilizer loop
ESTKALMAN: State out of bounds, resetting
SUP: Can not fly
SUP: Ready to fly
LHFL: Bitstream CRC32: 112BC794 [PASS]
LHFL: Firmware version 6 verified, booting deck!
ESTKALMAN: State out of bounds, resetting
[...]
```

I confirmed that OTA flashing the wifi-img-streamer example as shown above still works once all other decks are removed. Even just installing the LH deck (in addition to the AI deck, obviously) makes it stop at 3%; the same with all 3 decks attached.

Can you reproduce the issue on your end, e.g. with AI+LH decks and the latest master firmware you suggested?

**Reply by gemenerik** [reply]
OK, thanks for trying. I will try to reproduce and will keep you updated.

**Reply by gemenerik** [reply]
See https://github.com/bitcraze/aideck-gap8-examples/issues/105
